### PR TITLE
Target Android O

### LINKED ISSUE #415: Support Android adaptive icons
Since the Android O final release is coming up, it's recommended that Bitcoin Wallet supports [adaptive icons](https://developer.android.com/preview/features/adaptive-icons.html) to be consistent with other apps targeting O. I'd be happy to submit a PR.

### LINKED ISSUE #416: Support notification channels (Critical)
Apps targeting Android O [must support notification channels](https://developer.android.com/preview/features/notification-channels.html) or else notifications will not be shown. I propose adding a default notification channel for transactions which will include notifications for received bitcoin and user activity updates. Ideas for more notification channels are welcome.

EDIT: While I know this app isn't targeting O yet, even apps not targeting O that are running on O will have [reduced functionality](https://issuetracker.google.com/issues/62907528), so this is still somewhat important.

### PR DESCRIPTION
**Pull request by schildbach**
On my emulator, the custom sound for the notification channel doesn't work. If anyone got a real device with O can you test if you get the familiar ChingChing sound?

Fixes #415. Fixes #416.

### CODE REVIEW

File: wallet/src/de/schildbach/wallet/service/InactivityNotificationService.java [hunk 1]
```diff
@@ -46,7 +47,11 @@
  */
 public final class InactivityNotificationService extends IntentService {
     public static void startMaybeShowNotification(final Context context) {
-        context.startService(new Intent(context, InactivityNotificationService.class));
+        final Intent intent = new Intent(context, InactivityNotificationService.class);
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O)
```
Comment (TheCraftKid) [inline wallet/src/de/schildbach/wallet/service/InactivityNotificationService.java]: This `if` block can be replaced with ContextCompat.startForegroundService(context, intent). See [this](https://medium.com/google-developers/migrating-mediastyle-notifications-to-support-android-o-29c7edeca9b7).

File: wallet/src/de/schildbach/wallet/service/UpgradeWalletService.java [hunk 2]
```diff
@@ -38,7 +43,11 @@
  */
 public final class UpgradeWalletService extends IntentService {
     public static void startUpgrade(final Context context) {
-        context.startService(new Intent(context, UpgradeWalletService.class));
+        final Intent intent = new Intent(context, UpgradeWalletService.class);
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O)
```
Comment (TheCraftKid) [inline wallet/src/de/schildbach/wallet/service/UpgradeWalletService.java]: This if block can also be replaced with ContextCompat.startForegroundService(context, intent).

File: wallet/src/de/schildbach/wallet/service/InactivityNotificationService.java (hunk 1 above)
Comment (schildbach) [inline wallet/src/de/schildbach/wallet/service/InactivityNotificationService.java]: Thanks for your review! I'm using your suggestion now.

**Comment by schildbach** [comment]
I just enrolled one of my devices into the Android O beta program, so I could test the notification sound again. It works, seems as if the emulator has it wrong. I also could test the rounded rect launcher icon shape as that's what the default Nexus launcher is using.

File: wallet/src/de/schildbach/wallet/WalletApplication.java [hunk 3]
```diff
@@ -388,6 +396,31 @@ private void cleanupFiles() {
...
+                            .setUsage(AudioAttributes.USAGE_NOTIFICATION_EVENT).build());
+            nm.createNotificationChannel(received);
+
+            final NotificationChannel ongoing = new NotificationChannel(Constants.NOTIFICATION_CHANNEL_ID_ONGOING,
+                    getString(R.string.notification_channel_ongoing_name), NotificationManager.IMPORTANCE_LOW);
+            nm.createNotificationChannel(ongoing);
+
+            final NotificationChannel important = new NotificationChannel(Constants.NOTIFICATION_CHANNEL_ID_IMPORTANT,
+                    getString(R.string.notification_channel_important_name), NotificationManager.IMPORTANCE_HIGH);
+            nm.createNotificationChannel(important);
```
Comment (TheCraftKid) [inline wallet/src/de/schildbach/wallet/WalletApplication.java]: You could [batch](https://developer.android.com/reference/android/app/NotificationManager.html#createNotificationChannels(java.util.List<android.app.NotificationChannel>)) creating all these channels at once by calling `nm.createNotificationChannels(Arrays.asList(recieved, ongoing, important))`.

File: wallet/build.gradle [hunk 4]
```diff
@@ -35,7 +35,7 @@ ext {
 }
 
 android {
-    compileSdkVersion 'android-25'
+    compileSdkVersion 'android-26'
```
Comment (TheCraftKid) [inline wallet/build.gradle]: Shouldn't this just be `26`?

**Comment by TheCraftKid** [comment]
Sounds great! I'd suggest leaving the launcher icon as the pre-O one and then adding a new one in the resource directory mipmap-v26 to support adaptive icons. Nexus devices default to the square mask on Android O, but they use the non-adaptive versions otherwise.

File: wallet/src/de/schildbach/wallet/WalletApplication.java (hunk 3 above)
Comment (schildbach) [inline wallet/src/de/schildbach/wallet/WalletApplication.java]: Yes, but does it have an advantage? I think as long as the collection of channels isn't dynamic the single form is more appropriate.

File: wallet/build.gradle (hunk 4 above)
Comment (schildbach) [inline wallet/build.gradle]: We could indeed switch to the new format. Somehow the gradle plugin started with versions like this. But I'd put the switch into a different commit.

**Comment by schildbach** [comment]
@TheCraftKid The old icons are still there (ic_app_color_48dp) and used on pre-O. Maybe I should change the commit message to say "Add" rather than "Convert".

**Comment by TheCraftKid** [comment]
Is there an ETA for release?

**Comment by schildbach** [comment]
Currently its waiting for final Android O, which hopefully happens today.

**Comment by schildbach** [comment]
Merged.
